libethash-cl/kernels/cl/ethash.cl

```diff
@@ -233,10 +234,14 @@ do { \
     s = select(mix.s6, s, (x) != 6); \
     s = select(mix.s7, s, (x) != 7); \
     buffer[get_local_id(0)] = fnv(init0 ^ (a + x), s) % dag_size; \
-    mix = fnv(mix, g_dag[buffer[lane_idx]].uint8s[thread_id]); \
+    uint idx = buffer[lane_idx]; \
+    __global hash128_t const* g_dag; \
+    g_dag = (__global hash128_t const*) _g_dag0; \
+    if (idx & 1) \
+        g_dag = (__global hash128_t const*) _g_dag1; \
```

Comment:
```suggestion
   if (!(idx & 1)) \
     g_dag = (__global hash128_t const*) _g_dag0; \
    else \
        g_dag = (__global hash128_t const*) _g_dag1; \
```
This should save an address translation and semantically similar to same test in DAG generation

Reply:
You could do it that way but tried it and see no perceptible speed difference on 480.